src/Http/Headers/src/ContentDispositionHeaderValue.cs

```diff
@@ -618,54 +618,36 @@ private static bool TryDecodeMime(StringSegment input, [NotNullWhen(true)] out s
     private static string Encode5987(StringSegment input)
     {
         var builder = new StringBuilder("UTF-8\'\'");
-
-        var maxInputBytes = Encoding.UTF8.GetMaxByteCount(input.Length);
-        byte[]? bufferFromPool = null;
-        Span<byte> inputBytes = maxInputBytes <= MaxStackAllocSizeBytes
-            ? stackalloc byte[MaxStackAllocSizeBytes]
-            : bufferFromPool = ArrayPool<byte>.Shared.Rent(maxInputBytes);
-
-        var bytesWritten = Encoding.UTF8.GetBytes(input, inputBytes);
-        inputBytes = inputBytes[..bytesWritten];
-
-        int totalBytesConsumed = 0;
-        while (totalBytesConsumed < inputBytes.Length)
+        var remaining = input.AsSpan();
+        while (remaining.Length > 0)
         {
-            if (Ascii.IsValid(inputBytes[totalBytesConsumed]))
+            var length = remaining.IndexOfAnyExcept(Rfc5987AttrChar);
+            if (length < 0)
             {
-                // This is an ASCII char. Let's handle it ourselves.
-
-                char c = (char)inputBytes[totalBytesConsumed];
-                if (!AttrChar.Contains(c))
-                {
-                    HexEscape(builder, c);
-                }
-                else
-                {
-                    builder.Append(c);
-                }
-
-                totalBytesConsumed++;
+                length = remaining.Length;
             }
-            else
-            {
-                // Non-ASCII, let's rely on Rune to decode it.
+            builder.Append(remaining[..length]);
 
-                Rune.DecodeFromUtf8(inputBytes.Slice(totalBytesConsumed), out Rune r, out int bytesConsumedForRune);
-                Contract.Assert(!r.IsAscii, "We shouldn't have gotten here if the Rune is ASCII.");
+            remaining = remaining.Slice(length);
+            if (remaining.Length == 0)
+            {
+                break;
+            }
 
-                for (int i = 0; i < bytesConsumedForRune; i++)
-                {
-                    HexEscape(builder, (char)inputBytes[totalBytesConsumed + i]);
-                }
+            length = remaining.IndexOfAny(Rfc5987AttrChar);
+            if (length < 0)
+            {
+                length = remaining.Length;
+            }
 
-                totalBytesConsumed += bytesConsumedForRune;
+            for (var i = 0; i < length;)
+            {
+                Rune.DecodeFromUtf16(remaining.Slice(i), out Rune rune, out var runeLength);
```

Comment:
This was probably discussed elsewhere, but why did this change from Utf8 to Utf16?  Was the old code wrong?  Is the new code operating on different data?

Reply:
Ah, we did stop calling `Encoding.UTF8.GetBytes` and start operating on `input` directly.  Makes sense.

Reply:
Yes, that is removed, hence used Utf16 with the Rune.

Reply:
Yep, LGTM.  I just added the second comment in case anyone else was confused.